twake/frontend/src/app/components/ChatUploads/UploadsViewer.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+import { useChatUploadService } from 'app/state/recoil/hooks/useChatUploadService';
+import ChatUploadService from './ChatUploadService';
+import PendingFilesList from './PendingFileComponents/PendingFilesList';
+
+const ChatUploadsViewer = (): JSX.Element => {
+  const [pendingFilesListState] = useChatUploadService();
+
+  const currentTaskFiles = (pendingFilesListState || []).filter(
+    f =>
+      ChatUploadService.getPendingFile(f.id).uploadTaskId === ChatUploadService.currentTaskId ||
+      f.status === 'error',
+  );
+
```

Comment:
We should not have to mix hook and ChatUploadService method calls. If we have a hook, we must have the API to avoid calling the ChatUploadService singleton directly